### PR TITLE
Layer legend - Use the new option in config to display the legend image at startup

### PR DESCRIPTION
**Pull request by mdouchin**
Use the new option in Lizmap plugin dedicated to the layer legend image (in the layer tree)

* `auto`: same behaviour as before. The layer legend can be displayed by clicking on the arrow button
* `show`: show the legend image at startup
* `hide`: do not show the legend, nor the arrow to display it (as done before with the `noLegendImage` property)

* Funded by **Conseil Départemental du Gard (France)**

### CODE REVIEW

**Review by Gustry — APPROVED** [review]
LGTM

File: lizmap/www/assets/js/map.js [hunk 1]
```diff
@@ -168,6 +168,51 @@ var lizMap = function() {
...
+   * return fake versions for each property.
+   *
+   * Dependencies:
+   * config
+   */
+  function getLizmapDesktopPluginMetadata()
+  {
+    // Default fake versions if the properties does not yet exist in configuration file
+    var plugin_metadata = {
+      lizmap_plugin_version: 30108,
```
Comment (Gustry) [inline lizmap/www/assets/js/map.js:190]: Note for myself, I need to add a new key lizmap plugin version as int ...
In the CFG, it's like : 
```json
    "metadata": {
        "qgis_desktop_version": 31615,
        "lizmap_plugin_version": "3.7.7",
        "lizmap_web_client_target_version": 30500,
        "project_valid": true
    },
```

File: lizmap/www/assets/js/map.js [hunk 2]
```diff
@@ -1093,6 +1138,25 @@ var lizMap = function() {
...
+    } else {
+      var legendOption = 'auto';
+      if (nodeConfig.noLegendImage && nodeConfig.noLegendImage == 'True') {
+        legendOption = 'hide';
+      }
+    }
+
+    // Expand layer legend at startup
+    if (nodeConfig.type == 'layer' && legendOption == 'show') {
+      html += ' expanded ';
```
Comment (Gustry) [inline lizmap/www/assets/js/map.js:1157]:
```suggestion
    // Legend properties
    // Read the plugin metadata to get the legend options
    // depending on the configuration version
    let lizmap_plugin_metadata = getLizmapDesktopPluginMetadata();
    if (lizmap_plugin_metadata.lizmap_web_client_target_version >= 30600) {
      var legendOption = nodeConfig.hide_at_startup;
    } else {
      var legendOption = 'hide_at_startup';
      if (nodeConfig.noLegendImage && nodeConfig.noLegendImage == 'True') {
        legendOption = 'hide_at_startup';
      }
    }

    // Expand layer legend at startup
    if (nodeConfig.type == 'layer' && legendOption == 'expand_at_startup') {
      html += ' expanded ';
```

File: lizmap/www/assets/js/map.js [hunk 3]
```diff
@@ -1138,7 +1202,7 @@ var lizMap = function() {
     html += '</tr>';
 
     if (nodeConfig.type == 'layer'
-    && (!nodeConfig.noLegendImage || nodeConfig.noLegendImage != 'True')
+    && (legendOption != 'hide')
```
Comment (Gustry) [inline lizmap/www/assets/js/map.js:1205]:
```suggestion
    && (legendOption != 'hide_at_startup')
```

File: lizmap/www/assets/js/map.js [hunk 4]
```diff
@@ -1746,10 +1810,21 @@ var lizMap = function() {
...
 
+    // Read the plugin metadata to get the legend options
+    // depending on the configuration version
+    let lizmap_plugin_metadata = getLizmapDesktopPluginMetadata();
+    if (lizmap_plugin_metadata.lizmap_web_client_target_version >= 30600) {
+      var legendOption = nodeConfig.legendOption;
+    } else {
+      var legendOption = 'auto';
+      if (nodeConfig.noLegendImage && nodeConfig.noLegendImage == 'True') {
+        legendOption = 'hide';
```
Comment (Gustry) [inline lizmap/www/assets/js/map.js:1821]:
```suggestion
      var legendOption = 'hide_at_startup';
      if (nodeConfig.noLegendImage && nodeConfig.noLegendImage == 'True') {
        legendOption = 'disabled';
```

**Review by Gustry — CHANGES_REQUESTED** [review]

**Comment by Gustry** [comment]
New CFG in https://github.com/3liz/lizmap-plugin/pull/420 : 

* `legend_image_option` : 
  * `hide_at_startup`
  * `expand_at_startup`
  * `disabled`

Just to be a little bit more explicit, `hide` versus `disabled`.

**Comment by Gustry** [comment]
Test can be added after #3020 in the `tests/end2end/cypress/integration/legend-ghaction.js` file.

**Review by Gustry — APPROVED** [review]

**Review by rldhont — APPROVED** [review]

**Pull request merged**